js/controller.js

```diff
@@ -1073,7 +1073,16 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
     //get original video width/height dimensions
     getIntrinsicDimensions: function() {
       var video = this.state.mainVideoElement.get(0);
-      this.state.mainVideoAspectRatio = this.calculateAspectRatio(video.videoWidth, video.videoHeight);
+      var width, height;
+      if (typeof video.videoWidth != 'undefined'){
+        width = video.videoWidth;
+        height = video.videoHeight;
+      }
+      else {
```

Comment:
I should have mentioned this before - can we make the default HTML5?

Check for flash and if no flash default to html5.